modules/nf-core/semibin/singleeasybin/main.nf

```diff
@@ -4,43 +4,55 @@ process SEMIBIN_SINGLEEASYBIN {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/semibin:1.4.0--pyh7cba7a3_0':
-        'biocontainers/semibin:1.4.0--pyh7cba7a3_0' }"
+        'https://depot.galaxyproject.org/singularity/semibin:2.2.0--pyhdfd78af_0':
+        'biocontainers/semibin:2.2.0--pyhdfd78af_0' }"
 
     input:
     tuple val(meta), path(fasta), path(bam)
 
     output:
-    tuple val(meta), path("*.csv")                        , emit: csv
-    tuple val(meta), path("*.h5")                         , emit: model
-    tuple val(meta), path("output_prerecluster_bins/*.fa"), emit: output_fasta
-    tuple val(meta), path("output_recluster_bins/*.fa")   , emit: recluster_fasta
-    tuple val(meta), path("*.tsv")                        , emit: tsv
-    path "versions.yml"           , emit: versions
+    tuple val(meta), path("${prefix}/*.csv")              , emit: csv
+    tuple val(meta), path("${prefix}/*.h5")               , emit: model, optional: true
+    tuple val(meta), path("${prefix}/*.tsv")              , emit: tsv
+    tuple val(meta), path("${prefix}/output_bins/*.fa.gz"), emit: output_fasta
+    path "versions.yml", emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
-    def args3 = task.ext.args3 ?: ""
-    def prefix = task.ext.prefix ?: "${meta.id}"
+    def args  = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ""
+    prefix    = task.ext.prefix ?: "${meta.id}"
     """
 
-    SemiBin \\
+    SemiBin2 \\
         $args \\
         single_easy_bin \\
-        -i $fasta \\
-        -b $bam \\
-        -o $prefix \\
+        --input-fasta ${fasta} \\
+        --input-bam ${bam} \\
+        --output ${prefix} \\
         -t $task.cpus \\
-        $args3
+        $args2
 
-    mv $prefix/* .
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        SemiBin: \$( SemiBin2 --version )
+    END_VERSIONS
+    """
+    stub:
+    prefix    = task.ext.prefix ?: "${meta.id}"
+    """
+    mkdir ${prefix}
+    touch ${prefix}/{contig_bins,recluster_bins_info}.tsv
+    touch ${prefix}/{data,data_split}.csv
+    mkdir ${prefix}/output_bins
+    touch ${prefix}/output_bins/SemiBin_{0,1,2,3}.fa
+    gzip  ${prefix}/output_bins/SemiBin*
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
-        SemiBin: \$( SemiBin --version )
+        SemiBin: \$( SemiBin2 --version )
```

Comment:
I don't know if in nf-core we should allow different version of a tool to coexist as glimpse1 and 2.
But this is outside the scope of this PR

Reply:
The tool is called "SemiBin" on its github page, its just that they changed the executable name to be SemiBin2 when they made breaking changes. I think it wants to still be called SemiBin in the report.

Reply:
That's fine for me !